File: app/src/main/java/bogomolov/aa/fitrack/repository/dao/TagDao.java
package bogomolov.aa.fitrack.repository.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

import bogomolov.aa.fitrack.repository.entities.TagEntity;

@Dao
public interface TagDao {

    @Insert
    long insert(TagEntity tagEntity);

    @Delete
    void delete(TagEntity tagEntity);

    @Query("SELECT * from TagEntity")
    List<TagEntity> getTags();

}

File: app/src/main/java/bogomolov/aa/fitrack/repository/dao/TagDao.kt
package bogomolov.aa.fitrack.repository.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

import bogomolov.aa.fitrack.repository.entities.TagEntity

@Dao
interface TagDao {

    @get:Query("SELECT * from TagEntity")
    val tags: List<TagEntity>

    @Insert
    fun insert(tagEntity: TagEntity): Long

    @Delete
    fun delete(tagEntity: TagEntity)

}
